### ISSUE TITLE
CLI does not set exit code !=0 in case of validation errors

### ISSUE THREAD

**Issue by tomschr** [schxslt/schxslt#307]
This is a great project! Thank you very much for all your efforts! :heart: 

# Problem

If you run the CLI through  `schxslt-cli.jar` and get an error, the error isn't reflected in the return code. It is in all cases zero (tested on openSUSE 15.4, Java openjdk version "11.0.17" 2022-10-18, SchXslt CLI v1.9.4):

```
$ java -jar schxslt-cli.jar -v -s SCHEMATRON -d XMLFILE
[... some errors ...]
$ echo $?
0

$ java -jar schxslt-cli.jar -v -s SCHEMATRON -d XMLFILE
[... no errors ...]
$ echo $?
0
```

This is very unfortunate as you can't integrate the CLI command in a script and decide what to do if the validation creates an error or not. You need to parse the output which is cumbersome and error-prone.

# Proposal
On Unix/Linux it is a common practice to signal an error by returning an exit code that is _non zero_. If everything was ok it's zero. This helps to distinguish between these two cases and allows for integrate it into shell scripts.

If you want to go the extra mile (or kilometer), you could return different error codes to signal different problems. For example:

* error code `10`: no schematron schema found or readable
* error code `20`: no XML found or readable
* error code `30`: couldn't write output file
* error code `40`: couldn't find phase
* error code `200`: validation error

I'm not a Java programmer, but as far as I can see, it should be fairly easy. All you need is to call `System.exit(exit_code);` at the appropriate place. :wink:

**Comment by kthth** [comment]
I second this proposal. I'm using schxslt in a gitlab CI pipeline and need the return code to get notification of a failed job, i.e. invalid XML. Thanks

**Comment by dmj** [comment]
I reclassified it as a bug and so that I can put it in a v1.9.5 maintenance release, scheduled for this weekend.

**Comment by dmj** [comment]
Fix in v1.9.5. The commandline application now takes an argument -e --exitcode <n> that sets the exit code to be used for invalid documents.
